pandas/_libs/tslib.pyx

```diff
@@ -728,7 +728,7 @@ class Timestamp(_Timestamp):
         # reconstruct & check bounds
         ts_input = datetime(dts.year, dts.month, dts.day, dts.hour, dts.min,
                             dts.sec, dts.us, tzinfo=_tzinfo)
-        ts = convert_to_tsobject(ts_input, _tzinfo, None, 0, 0)
+        ts = _convert_datetime_to_tsobject(ts_input, _tzinfo)
```

Comment:
no need to break with spelling convention. call ``convert_from_datetime_to_tsobject``